common/f3d_materials.md

```diff
@@ -0,0 +1,193 @@
+
+```{card} Texture Gen Conversions
+:text-align: center
+{math}`Gen(-1, 0) \rightarrow (0, 0.5) \hspace{1cm} GenLinear(-1, 0)\rightarrow (0, 0.5)`
+{math}`Gen(-\sqrt{2}, \sqrt{2}) \rightarrow (-\sqrt{2}, \sqrt{2}) \hspace{1cm} GenLinear(-\sqrt{2}, \sqrt{2})\rightarrow (0.25, 0.75)`
+{math}`Gen(0, 1) \rightarrow (0.5, 1) \hspace{1cm} GenLinear(0, 1)\rightarrow (0.5, 1)`
+```
+
+## Smooth Shading
+
+With Shading is enabled, then [Smooth Shading]{.key_text} will interpolate between the vertices of each triangle to create a smooth color. This is typically known as Gouraud Shading.
+
+::::
+::::{tab-item} Upper/Lower
+
+The upper and lower tabs feature a slew of various options for unique features the RCP is capable of performing. These features offer core control over how rendering occurs on the N64, and some options have considerable depth.
+
+## RGB/Alpha Dither
+
+Dithering affects the final color that is output to the framebuffer. The affect is used to reduce artifacts and banding when converting higher precision source colors from varoius sources into the lower precision framebuffer. 8 bits are used per RGB component in the RDP pipeline, and it is written to a 5 bit per channel framebuffer. Try changing the types of dithering if banding occurs.
+
+* Pattern - Alpha only option. Follows the same dithering as rgb or uses Bayer
+* NotPattern - Alpha only option. Dithers with the opposite pattern as "Pattern"
+* Magic Square - RGB only option. Uses a magic square for dithering
+* Bayer - RGB only option. Uses a Bayer matrix for dithering
+* Noise - Random dithering
+* Enable - Option just for backwards compatibility. Equivalent to "Noise"
+* Disable - No dithering is uses.
+
+## Chroma Key
+
+Chroma keying allows you to specify colors as pixel alpha. A combiner with equation {eq}`chroma_key_combiner_f3d_mat` is recommended for use and should be used in the last available cycle. The combiner will use the `A` value put in variable A as the color output, and the alpha out will be the min value from each color channel of {eq}`key_alpha_calc_f3d_mat`.
+In order to use chroma key, `Key` must be selected in the chroma key dropdown and [alpha_cvg_sel]{.key_text} must be off in the chosen render mode.
+
+:::{card}
+$$
+combined\_color = (A - key\_center)*key\_scale + 0
+$$(chroma_key_combiner_f3d_mat)
+$$
+key\_alpha = clamp(0, key\_width - abs(combined\_color), 255)
+$$(key_alpha_calc_f3d_mat)
+:::
+
+## Texture Convert/Filter
+
+Texture Convert allows for the enabling of YUV conversion, and texture filtering. Filtering type is controlled by the Texture Filter option. For filtering, billinear is standard, and point is for no filtering. Average is a special type of filter only for pixel/texel aligned rendering.
```

Comment:
point filtering (aka nearest neighbour) is still a type of filtering, it's not "no filtering"

```suggestion
Texture Convert allows for the enabling of YUV conversion, and texture filtering. Filtering type is controlled by the Texture Filter option. For filtering, bilinear is standard. Point filtering is also known as "nearest neighbour". Average is a special type of filter only for pixel/texel aligned rendering.
```